.github/workflows/workflow.yml

```diff
@@ -1,57 +1,85 @@
 name: Main workflow
 on: [push, pull_request]
 jobs:
-  run:
-    name: Run
+  build:
     runs-on: ${{ matrix.operating-system }}
     strategy:
       matrix:
         operating-system: [ubuntu-latest, windows-latest]
     steps:
     - name: Checkout
       uses: actions/checkout@v2
-
-    - name: Setup Node.js 12.x
+    - name: Setup Node.js 12
       uses: actions/setup-node@v1
       with:
         node-version: 12.x
-
     - name: npm install
       run: npm install
-
     - name: Lint
       run: npm run format-check
-
     - name: npm test
       run: npm test
-  run-with-proxy:
-    name: Run with proxy
-    runs-on: ubuntu-latest
-    services:
-      squid:
-        image: dakale/squid
-        ports: ['3128:3128']
-        options: '--health-cmd "exit 0" --health-interval 3s'
-    env:
-      http_proxy: http://localhost:3128
-      https_proxy: http://localhost:3128
+
+  test:
+    runs-on: ${{ matrix.operating-system }}
+    strategy:
+      matrix:
+        operating-system: [ubuntu-latest, windows-latest]
     steps:
-    - name: Block non proxied traffic
-      run: |
-        echo "127.0.0.0 registry.npm.js nodejs.org github.com api.github.com download.java.net static.azul.com" | sudo tee -a /etc/hosts
     - name: Checkout
       uses: actions/checkout@v2
-
-    - name: Setup Node.js 12.x
-      uses: actions/setup-node@v1
+    - name: Clear tool cache
+      if: runner.os != 'windows'
+      run: mv "${{ runner.tool_cache }}" "${{ runner.tool_cache }}.old"
+    - name: Clear tool cache (Windows)
+      if: runner.os == 'windows'
+      run: move "${{ runner.tool_cache }}" "${{ runner.tool_cache }}.old"
+    - name: Setup Java 13
+      uses: ./
       with:
-        node-version: 12.x
+        java-version: 13.0.2
+    - name: Verify Java 13
+      if: runner.os != 'windows'
+      run: __tests__/verify-java.sh 13.0.2
+    - name: Verify Java 13 (Windows)
+      if: runner.os == 'windows'
+      run: __tests__/verify-java.ps1 13.0.2
 
-    - name: npm install
-      run: npm install
-
-    - name: Lint
-      run: npm run format-check
+  test-proxy:
```

Comment:
Switched the proxy test to:
1. Run in a container, with dns disabled (`options: --dns 127.0.0.`). This guarantees that all requests are actually going through the proxy. Previous approach hacked hosts file which does not guarantee all traffic.
2. Switched to widely used squid container image, rather than maintain our own
3. Switched to testing `./` during proxy test rather than unit tests